util/network-devp2p/src/discovery.rs

```diff
@@ -518,7 +518,14 @@ impl<'a> Discovery<'a> {
 
 	fn on_ping(&mut self, rlp: &Rlp, node_id: &NodeId, from: &SocketAddr, echo_hash: &[u8]) -> Result<Option<TableUpdates>, Error> {
 		trace!(target: "discovery", "Got Ping from {:?}", &from);
-		let ping_from = NodeEndpoint::from_rlp(&rlp.at(1)?)?;
+		let ping_from = if let Ok(rlp) = rlp.at(1) {
+			NodeEndpoint::from_rlp(&rlp)?;
```

Comment:
Is this correct? I think it will always return `Ok` since we have more elements in the RLP, but will fail on creating `NodeEndpoint` since the values are `null`.

Any chance we can have a test for this?

Reply:
Yep, it should handle the `Err` case of `NodeEndpoint::from_rlp`, not the one of `rlp.at(1)`.